# Statistics: Confidence Intervals


In [1]:
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
# EXERCISE 1. 
# Suppose scores on exams in statistics are normally distributed with an unknown population mean and a population standard deviation of 3 points. 
# A random sample of 36 scores is taken and gives a sample mean (sample  mean score) of 68. 
# Find a confidence interval estimate for the population mean exam score (the mean score on all exams).
# Find a 90% confidence interval for the true (population) mean of statistics exam scores.

In [3]:
x_bar = 68 # sample mean 
n = 36 # number of samples
sigma = 3 # sd of population
#mu =  # Population mean

steps to calculate confidence intervals (90%, 95%, 99%)
- idenitfy sample mean
- identify z-score
- identify standard deviation
- identify the sample mean

In [7]:
#Calculate the test statistic
z =1.645 #zscore at the 90% CI

In [8]:
#calculate standard error of the mean
sem = sigma / np.sqrt(n)
sem

0.5

In [9]:
moe = 1.645 * sem
moe

0.8225

In [10]:
upper_lim= 68 + moe
upper_lim

68.8225

In [11]:
low_lim= 68 - moe
low_lim

67.1775


#Results: 90% Confidence Interval: 68 ± 0.822; (67.2 to 68.8)

In [ ]:
# EXERCISE 2. 
# What is the normal body temperature for healthy humans? 
# A random sample of 130 healthy human body temperatures provided by Allen Shoemaker7 
# yielded 98.25 degrees and standard deviation 0.73 degrees. 
# Give a 99% confidence interval for the average body temperature of healthy people.

In [12]:
x_bar = 98.25 # sample mean 
n = 130 # number of samples
sigma = 0.73 # sd of population
#mu =     # Population mean

In [13]:
#Calculate the test statistic
z =2.576 #zscore at the 99% CI

In [14]:
#calculate standard error of the mean
sem = sigma / np.sqrt(n)
sem

0.06402523540941313

In [20]:
moe = 2.576 * sem
moe

0.16492900641464822

In [21]:
upper_lim= 98.25 + moe
upper_lim

98.41492900641465

In [22]:
low_lim= 98.25 - moe
low_lim

98.08507099358535

#Results: "With 99% confidence, the normal body temperature for healthy humans is between 98.1 and 98.4, based on 130 samples."

In [ ]:
# EXERCISE 3. 
# The administrators for a hospital wished to estimate the average number of days required for inpatient 
# treatment of patients between the ages of 25 and 34. 
# A random sample of 500 hospital patients between these ages produced a mean and standard deviation equal to 5.4 and 3.1 days, respectively.
# Construct a 95% confidence interval for the mean length of stay for the population of patients from which the sample was drawn.

In [ ]:
x_bar = 5.4 # sample mean 
n = 500 # number of samples
sigma = 3.1 # sd of population
#mu =  # Population mean

In [23]:
#Calculate the test statistic
z =1.960 #zscore at the 95% CI

In [24]:
#calculate standard error of the mean
sem = sigma / np.sqrt(n)
sem

0.06402523540941313

In [25]:
moe = 1.960 * sem
moe

0.12548946140244974

In [26]:
upper_lim= 5.4 + moe
upper_lim

5.52548946140245

In [27]:
low_lim= 5.4 - moe
low_lim

5.274510538597551

#Results: "With 95% confidence, the mean length of stay for the population of patients is between 5.3 and 5.5, based on 500 samples."